### PR TITLE
Update samples readme

### PR DESCRIPTION
**Pull request by BillWagner**
This PR adds readme.md files for all samples, and updates a few terms in some of the LINQ samples.

### CODE REVIEW

File: docs/tutorials/getting-started-with-csharp/WeatherMicroservice/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+================
+
+This sample is created during the [Creating Microservices hosted in Docker](../microservices.html)
+tutorial. Please see that topic for detailed steps on the code
+for this sample.
+
+Key Features
+------------
+
+This sample demonstrates creating a web service using asp.net core and C#
```
Comment (svick) [inline docs/tutorials/getting-started-with-csharp/WeatherMicroservice/README.md]: Capitalization: ASP.NET Core

File: docs/tutorials/getting-started-with-csharp/WeatherMicroservice/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+C# Microservices Sample
+================
+
+This sample is created during the [Creating Microservices hosted in Docker](../microservices.html)
```
Comment (svick) [inline docs/tutorials/getting-started-with-csharp/WeatherMicroservice/README.md]: This is just a project README that won't be published to the docs website, right?

In that case, this (and all the other links) should probably be absolute links to the website.

File: samples/csharp-language/indexers/README.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+This sample is created during the [Indexers](../../docs/languages/csharp/indexers.html)
+for learning C# features. Please see that topic for detailed steps on the code
+for this sample.
+
+Key Features
+------------
+
+This sample contains for different samples involving
+indexers. You'll learn different C# idioms where indexers can be
+used to expression your design intent.
```
Comment (svick) [inline samples/csharp-language/indexers/README.md]: typo: expression → express?

File: samples/csharp-language/delegates-and-events/README.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+C# Delegates and EVents Sample
```
Comment (cartermp) [inline samples/csharp-language/delegates-and-events/README.md]: Spelling: EVents ---> Events

**Comment by mairaw** [comment]
LGTM modulo the feedback already given

File: docs/tutorials/getting-started-with-csharp/WeatherMicroservice/README.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
+C# Microservices Sample
```
Comment (richlander) [inline docs/tutorials/getting-started-with-csharp/WeatherMicroservice/README.md]: This file uses mixed case, where the other ones use all lower case.

File: docs/tutorials/getting-started-with-csharp/WeatherMicroservice/README.md (hunk 2 above)
Comment (richlander) [inline docs/tutorials/getting-started-with-csharp/WeatherMicroservice/README.md]: Two thoughts.

- I'd first introduce the sample for what it is, not the fact that it is associated with some content. The "key features" content seems like the intro to me. I'd then follow up with the tutorial link.
- "during" seems like an odd term to describe its association. I was expecting something for like: "You can read the [Underwater basket weaving made easy](https://underwaterbasketweaving.com) tutorial for step-by-step instructions on creating this sample."

File: docs/tutorials/getting-started-with-csharp/WeatherMicroservice/README.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+for this sample.
+
+Key Features
+------------
+
+This sample demonstrates creating a web service using asp.net core and C#
+and deploying that service in a Docker Container.
+
+Build and Run
+-------------
```
Comment (richlander) [inline docs/tutorials/getting-started-with-csharp/WeatherMicroservice/README.md]: Each readme would benefit from a single link to first install the .NET Core SDK.

File: samples/getting-started/golden/README.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
+Getting Started using Visual Studio Code Sample
```
Comment (richlander) [inline samples/getting-started/golden/README.md]: we should rename "golden" to something else.

**Comment by richlander** [comment]
There seems to be quite a mix of top-level folders. Do we have a coherent approach to that? It would be good to include some guidance somewhere on the use of top-level folders for new contributors. That would help us validate if we're happy with the current use.

**Comment by BillWagner** [comment]
@richlander Yes, there is a mix of top level folders. That will be a separate PR. I'm just trying to avoid too many crazy merges.

**Comment by richlander** [comment]
Let's get this in and we can resolve other issues later.

**Comment by BillWagner** [comment]
@richlander I'll address the comments on this one as I do the moving of samples for the next PR

**Pull request merged**